### PR TITLE
FilterTable syntax update

### PR DESCRIPTION
**Pull request by TrevorBramble**
Convert FilterTable to new syntax and make requisite dependency requirement change.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ source 'https://rubygems.org'
 
 gem 'faraday',            '~> 0.15.0'
 gem 'faraday_middleware', '~> 0.12.2'
-gem 'inspec'
+gem 'inspec',             '~> 2.2',  '>= 2.2.7'
```
Comment (dmccown) [inline Gemfile]: 👍 Thanks!

File: libraries/azurerm_resource.rb [hunk 2]
```diff
@@ -34,3 +30,11 @@ def credentials
     end
   end
 end
+
+class AzurermPluralResource < AzurermResource; end
+
+class AzurermSingularResource < AzurermResource
+  def exists?
```
Comment (dmccown) [inline libraries/azurerm_resource.rb]: @clintoncwolfe can you recommend if we should do `exist` or `exists`?

Comment (TrevorBramble) [reply libraries/azurerm_resource.rb]: @dmccown `exists?` is preferred, and I assume `exist?` is a fallback. https://github.com/inspec/inspec/blob/master/docs/dev/filtertable-usage.md#some-things-are-defined-for-you

Comment (clintoncwolfe) [reply libraries/azurerm_resource.rb]: FilterTable, used for plural resources, uses the plural inflection, `exist?` 

Singular resources should use the singular inflection, `exists?`.

That's my take.

**Pull request merged**